doc/source/roadmap.rst

```diff
@@ -1 +1,104 @@
+some point).  What we want is sparse *arrays* that act like ``numpy.ndarray``.
+This is being worked on in https://github.com/pydata/sparse, which is quite far
+along.  The tentative plan is:
+
+- Start depending on ``pydata/sparse`` once it's feature-complete enough (it
+  still needs a CSC/CSR equivalent) and okay performance-wise.
+- Add support for ``pydata/sparse`` to ``scipy.sparse.linalg`` (and perhaps to
+  ``scipy.sparse.csgraph`` after that).
+- Indicate in the documentation that for new code users should prefer
+  ``pydata/sparse`` over sparse matrices.
+- When NumPy deprecates ``numpy.matrix``, vendor that or maintain it as a
+  stand-alone package.
+
+
+Fourier transforms
+------------------
+
+We want to integrate PocketFFT into ``scipy.fftpack`` for significant
+performance improvements (see `this NumPy PR
+<https://github.com/numpy/numpy/pull/11888>`__ for details),
+add a backend system to support PyFFTW and mkl-fft,
+and align the function signatures of ``numpy.fft`` and ``scipy.fftpack``.
+
+
+Support for distributed arrays and GPU arrays
+---------------------------------------------
+
+NumPy is splitting its API from its execution engine with
+``__array_function`` and ``__array_ufunc__``.  This will enable parts of SciPy
+to accept distributed arrays (e.g. ``dask.array.Array``) and GPU arrays (e.g.
+``cupy.ndarray``) that implement the ``ndarray`` interface.  At the moment it is
+not yet clear which algorithms will work out of the box, and if there are
+significant performance gains when they do.  We want to create a map of which
+parts of the SciPy API work, and improve support over time.
+
+
+Improve benchmark system for ``optimize``
+-----------------------------------------
+
+``scipy.optimize`` has an extensive set of benchmarks for accuracy and speed of
+the global optimizers. That has allowed adding new optimizers (``shgo`` and
+``dual_annealing``) with significantly better performance than the existing
+ones.  The benchmark system itself is slow and hard to use however; we need to
+make it faster and make it easier to compare performance of optimizers via
+plotting performance profiles.
+
+
+Linear programming enhancements
+-------------------------------
+
+Recently all known issues with ``optimize.linprog`` have been solved.  Now we
+have many ideas for additional functionality (e.g. integer constraints, sparse
+matrix support, performance improvements), see gh-9269.
+
+
+Improve source builds on Windows
+--------------------------------
+
+SciPy critically relies on Fortran code. This is still problematic on Windows.
+There are currently only two options: using Intel Fortran), or using
+MSVC + gfortran.  The former is expensive, while the latter works (it's what we
+use for releases) but is quite hard to do correctly.  For allowing contributors
+and end users to reliably build SciPy on Windows, using the Flang compiler
+looks like the best way forward long-term.  Until Flang support materializes,
```

Comment:
Cool, I'm not familiar with `Flang` but definitely agree this stuff is painful as I've worked through the Azure stuff for SciPy  windows builds (64-bit builds just starting to work after what seemed like days of guesswork!).